pika/adapters/blocking_connection.py

```diff
@@ -344,6 +342,8 @@ def _handle_read(self):
         keeps track of how many frames have been written since the last read.
 
         """
+        if not hasattr(self, '_read_poller'):
+            raise exceptions.ConnectionClosed()
```

Comment:
@vitaly-krugl, I think we're talking about different things. I _think_ I've tested your PRs and assuming I've pulled them correctly I don't think they address the issue.

Test case below....

```python
import pika
import logging

class PubSub:

  def __init__(self, url='amqp://guest:guest@localhost:5672/%2F'):
    self.conn = pika.BlockingConnection(pika.URLParameters(url))
    self.chan = self.conn.channel()
    # works more reliably if we make sure the q is empty
    self.chan.queue_delete(queue='testq') 
    self.chan.queue_declare(queue='testq')
    self.chan.queue_bind(queue='testq',exchange='amq.direct',routing_key='test')
    self.chan.basic_consume(self.callback,queue='testq')
    self.count = 0

  def run(self):
    while True:
        try:
            logging.info("Start consuming.....")
            self.chan.start_consuming()

        except pika.exceptions.ConnectionClosed:
            logging.error("WTF?!, my connection got closed")
            raise
       
        except Exception as e:
            logging.error("start_consuming raised: %s", e)
  
 
  def callback(self, cha, meth, header, body):
    if self.count == 0:
        self.count += 1
        logging.info("In callback, about to generate some generic exception")
        a = undefined_var
    else:
        logging.info("In callback, about to generate an AttributeError")
        a = body.some_random_attribute

  def publish(self, body):
    self.chan.basic_publish('amq.direct','test', body)

logging.basicConfig(level=logging.INFO)

p = PubSub()
p.publish('message1') 
p.publish('message2') 
p.run()
```

With the broken code, output is...

```
INFO:pika.adapters.base_connection:Connecting to ::1:5672
INFO:root:Start consuming.....
INFO:root:In callback, about to generate some generic exception
ERROR:root:start_consuming raised: global name 'undefined_var' is not defined
INFO:root:Start consuming.....
INFO:root:In callback, about to generate an AttributeError
ERROR:root:WTF?!, my connection got closed
Traceback (most recent call last):
  File "attribute_error_test.py", line 47, in <module>
    p.run()
  File "attribute_error_test.py", line 20, in run
    self.chan.start_consuming()
  File "/home/will/git/vitaly-krugl-pika/pika/adapters/blocking_connection.py", line 965, in start_consuming
    self.connection.process_data_events()
  File "/home/will/git/vitaly-krugl-pika/pika/adapters/blocking_connection.py", line 243, in process_data_events
    raise exceptions.ConnectionClosed()
pika.exceptions.ConnectionClosed
```
Should be...
```
INFO:pika.adapters.base_connection:Connecting to ::1:5672
INFO:root:Start consuming.....
INFO:root:In callback, about to generate some generic exception
ERROR:root:start_consuming raised: global name 'undefined_var' is not defined
INFO:root:Start consuming.....
INFO:root:In callback, about to generate an AttributeError
ERROR:root:start_consuming raised: 'str' object has no attribute 'some_random_attribute'
INFO:root:Start consuming.....
```